tests/Feature/LMSJoinClassTest.php

```diff
@@ -0,0 +1,105 @@
+<?php
+
+namespace Uasoft\Badaso\Module\LMSModule\Tests\Feature;
+
+use Uasoft\Badaso\Module\LMSModule\Models\Course;
+use Uasoft\Badaso\Module\LMSModule\Models\CourseUser;
+use Uasoft\Badaso\Module\LMSModule\Models\User;
+use Tests\TestCase;
+use Uasoft\Badaso\Module\LMSModule\Tests\Helpers\AuthHelper;
+
+class LMSJoinClassTest extends TestCase
+{
+
+  public function testJoinClassWithoutLoginExpectResponseStatus401()
+  {
+    $url = route('badaso.course.join');
+    $response = $this->json('POST', $url);
+    $response->assertStatus(401);
+  }
+
+  public function testJoinClassAsAuthorizedUserWithUnknownClassCodeExpectResponseStatus500()
+  {
+      $user = User::factory()->create();
+      $user->rawPassword = 'password';
+
+      $url = route('badaso.course.join');
+
+      $response = AuthHelper::asUser($this, $user)->json('POST', $url, [
+          'code' => 'xxx'
+      ]);
+      $response->assertStatus(500);
+  }
```

Comment:
I think this one should return 404 or 400 instead, since unknown class code should not be an issue in our server